File: src/main/java/org/c3lang/intellij/psi/C3CallablePsiElement.java
package org.c3lang.intellij.psi;

import com.intellij.navigation.NavigationItem;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface C3CallablePsiElement extends C3FullyQualifiedNamePsiElement, NavigationItem
{
	@NotNull String getSourceFileName();
	@Nullable ShortType getType();
	@Nullable ShortType getReturnType();
	@NotNull List<ParamType> getParameterTypes();
}

File: src/main/java/org/c3lang/intellij/psi/C3CallablePsiElement.kt
package org.c3lang.intellij.psi

interface C3CallablePsiElement : C3PsiElement {
    val sourceFileName: String
    val moduleName: ModuleName?
    val typeName: TypeName?
    val functionOrMacroName: FullyQualifiedName
    val returnTypeName: ReturnTypeName?
    val parameterTypeNames: String
//    val parameterTypeNames: List<ParameterTypeName>
}